internal/quantile/aggregate.go

```diff
@@ -74,36 +74,43 @@ func (e *E2eProcessingLatencyAggregate) Less(i, j int) bool {
 	return e.Percentiles[i]["percentile"] > e.Percentiles[j]["percentile"]
 }
 
+// Add merges two E2eProcessingLatencyAggregate by averaging the percentiles
+//
+// It is side-effect free and does not modify the receiver or the parameter
```

Comment:
No, Go is pass-by-value always so that's not possible.

My suggestion is to change the calling code so that the receiver is never nil. I'm happy to send over a counter-PR to fix this bug, if you like.